subsys/net/lib/sockets/Kconfig

```diff
@@ -246,18 +246,39 @@ config NET_SOCKETS_CAN_RECEIVERS
 config NET_SOCKETPAIR
 	bool "Support for socketpair"
 	select PIPES
-	depends on HEAP_MEM_POOL_SIZE != 0
 	help
 	  Communicate over a pair of connected, unnamed UNIX domain sockets.
 
+if NET_SOCKETPAIR
+
 config NET_SOCKETPAIR_BUFFER_SIZE
 	int "Size of the intermediate buffer, in bytes"
 	default 64
 	range 1 4096
-	depends on NET_SOCKETPAIR
 	help
 	  Buffer size for socketpair(2)
 
+choice
+	prompt "Memory management for socketpair"
+	default NET_SOCKETPAIR_HEAP if HEAP_MEM_POOL_SIZE != 0
+
+config NET_SOCKETPAIR_STATIC
+	bool "Pre-allocate memory statically"
+
+config NET_SOCKETPAIR_HEAP
+	bool "Use heap for allocating socketpairs"
+	depends on HEAP_MEM_POOL_SIZE != 0
+
+endchoice
+
+if NET_SOCKETPAIR_STATIC
+config NET_SOCKETPAIR_MAX
+	int "How many socketpairs to pre-allocate"
+	default 1
+	range 1 16
```

Comment:
If we map this to 32bit var, then minimum is to check that if this value is changed, there will be a build error, so please add BUILD_ASSERT somewhere to enforce this.

Also we could allow arbitrary long bitmasks with ATOMIC_DEFINE and not rely on int size. Something like this could be implemented later.

Reply:
sys_bitarray is also useful for this sort of thing. I would prefer if the locking of the sys_bitarray were external though.